packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts

```diff
@@ -0,0 +1,112 @@
+import type {
+  CommonLogger,
+  RedisConfig,
+  TransactionObservabilityManager,
+} from '@lokalise/node-core'
+import Redis from 'ioredis'
+import { QUEUE_IDS_KEY } from '../constants'
+import { BackgroundJobProcessorLogger } from '../logger/BackgroundJobProcessorLogger'
+import type {
+  BackgroundJobProcessorConfig,
+  BackgroundJobProcessorDependencies,
+} from '../processors/types'
+import type { BaseJobPayload, RequestContext, SafeJob } from '../types'
+import { resolveJobId, sanitizeRedisConfig } from '../utils'
+
+const queueIdsSet = new Set<string>()
+
+export class BackgroundJobProcessorMonitor<
```

Comment:
Great job on this abstraction! It's a really well chosen cohesive unit of related functionality, and separating it does make code easier to understand.